packages/client/lib/execution/vmexecution.ts

```diff
@@ -50,7 +51,7 @@ export class VMExecution extends Execution {
 
     if (this.config.vm === undefined) {
       const trie = new Trie({
-        db: new LevelDB(this.stateDB),
+        db: new LevelDB(this.stateDB) as DB,
```

Comment:
It is in this instance since the `client` DB allows more key and value types than does the `trie`.

Reply:
Can we optimally align this so that we have one single interface for Trie, Blockchain and eventual other usages? 

(this goes a bit together with the special-handling topic for the encoding values)